sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/MethodOverloadsShouldBeGrouped.vb

```diff
@@ -0,0 +1,134 @@
+'                            ^^^^
+        End Sub
+
+        Public Overloads Function Test2() ' Noncompliant {{All 'test2' signatures should be adjacent}}
+'                                 ^^^^^
+        End Function
+
+        Public Overloads Function Test2(ByVal a As Integer)
+        End Function
+
+        Private Class NestedClass
+        End Class
+
+        Public Overloads Function Test2(ByVal a As Integer, ByVal b As Integer) ' Secondary {{Non-adjacent overload}}
+'                                 ^^^^^
+        End Function
+
+        Public Overloads Function Test3() ' Noncompliant {{All 'test3' signatures should be adjacent}}
+        End Function
+
+        Public Overloads Function Test3(ByVal a As Integer)
+        End Function
+
+        Public Age As Integer
+
+        Public Overloads Function test3(ByVal a As Double) ' Secondary {{Non-adjacent overload}}
+        End Function
+
+        Public Overloads Function Test4()
+        End Function
+
+        Public Overloads Function Test4(ByVal a As Integer)
+        End Function
+
+        Public Overloads Function Test4(ByVal a As Double)
+        End Function
+
+        Public Overloads Function TEST3(ByVal a As Double, ByVal b As Integer) ' Secondary {{Non-adjacent overload}}
+        End Function
+    End Class
+
+    Class B
+        Public Sub New() ' Noncompliant {{All 'new' signatures should be adjacent}}
+        End Sub
+
+        Public Sub New(ByVal x As Integer)
+        End Sub
+
+        Protected Overrides Sub Finalize()
+        End Sub
+
+        Public Sub New(ByVal x As Double) ' Secondary {{Non-adjacent overload}}
+        End Sub
+    End Class
+
+    Class C
+        Public Overloads Sub Test()
+        End Sub
+
+        Public Overloads Sub TEST(ByVal a As Double)
+        End Sub
+
+        Public Overloads Sub Test(ByVal a As Integer)
+        End Sub
+
+        Public Overloads Function Test2()
+        End Function
+
+        Public Overloads Function Test2(ByVal a As Integer)
+        End Function
+    End Class
+
+    Interface D
+        Function TEST() As Integer ' Noncompliant {{All 'test' signatures should be adjacent}}
+
+        Function test(ByVal a As Double) As Integer
+
+        Event MyEvent(ByVal Success As Boolean)
+
+        Function TEST2() As Integer
+
+        Function tEst2(ByVal a As Integer) As Integer
+
+        Function test2(ByVal a As Double) As Integer
+
+        Function tEst(ByVal a As Integer) As Integer ' Secondary {{Non-adjacent overload}}
+
+        Event MyEvent2(ByVal Success As Boolean)
+    End Interface
+
+    Interface E
+        Function TEST() As Integer
+
+        Function test(ByVal a As Double) As Integer
+
+        Function tEst(ByVal a As Integer) As Integer
+
+        Event MyEvent2(ByVal Success As Boolean)
+    End Interface
+
+    Structure F
+        Public myField1 As String
+
+        Public Sub Test() ' Noncompliant {{All 'test' signatures should be adjacent}}
```

Comment:
I would have tried to not use the lower case representation `test` of the method name in the message, but the first occurrence `Test`. For example, if the method name is `AMethodNameEasyToRead`, it's not nice to display `amethodnameeasytoread`.